arithmetization/src/main/java/net/consensys/linea/zktracer/module/blockdata/Blockdata.java

```diff
@@ -93,7 +94,20 @@ public List<ColumnHeader> columnsHeaders() {
   @Override
   public void commit(List<MappedByteBuffer> buffers) {
     final long firstBlockNumber = operations.getFirst().absoluteBlockNumber();
-    final long chainId = getChainIdFromTransaction(rlpTxn.operations().getLast().tx());
+
+    // TODO: this doesn't work if all transaction of the batch are WO ChainId
+    long chainId = -1;
+    for (RlpTxnOperation tx : rlpTxn.operations().getAll()) {
+      try {
+        chainId = getChainIdFromTransaction(tx.tx());
+      } catch (Exception e) {
+        continue;
+      }
+      if (chainId != -1) {
```

Comment:
If the call succeeds in the try block, then chainId cannot be -1 right? So i think you can just return.

Reply:
indeed